### PR TITLE
TELCODOCS-1184: Configuring vSphere Connection Details

### PR DESCRIPTION
**Pull request by tshwartz**
Version(s):
Main, 4.13, 4.12

Link to issue: https://issues.redhat.com/browse/TELCODOCS-1184

Docs preview:
Reference to new section: https://57799--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/preparing-to-install-on-vsphere.html
New section: https://57799--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-post-installation-configuration.html
Link to Google Doc version with questions (outdated): https://docs.google.com/document/d/1ES2Of_f584bO_D6o3msBYr-9kybEh3bVrnI4X-ZZ8s4/edit#

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by vrutkovs — APPROVED** [review]

File: installing/installing_vsphere/installing-vsphere-post-installation-configuration.adoc
Comment (mareklibra) [inline installing/installing_vsphere/installing-vsphere-post-installation-configuration.adoc:8]: Instead of `invalid`, what about denoting the fact that they are placeholders? Logically `invalid` is correct, but sounds strong or as a bug to me.

File: modules/configuring-vsphere-connection-details.adoc
Comment (mareklibra) [inline modules/configuring-vsphere-connection-details.adoc:28]: Both the `username` or `password` should be provided encoded, the real human-readable values. The base64 envoding is up to the application to do (plugin)

File: modules/configuring-vsphere-connection-details.adoc
Comment (mareklibra) [inline modules/configuring-vsphere-connection-details.adoc:37]: It's not a list to choose from (not a dropdown). It's just a regular text-input field, so the user types the value there.

File: modules/configuring-vsphere-connection-details.adoc
Comment (mareklibra) [inline modules/configuring-vsphere-connection-details.adoc:44]: Not a list to choose from, the user needs to type the value. There's hint how to compose it.

File: modules/configuring-vsphere-connection-details.adoc
Comment (mareklibra) [inline modules/configuring-vsphere-connection-details.adoc:45]: It's not `config.map` file.

Its `ConfigMap` (one of the Kubernetes resource types), it's name is `cloud-provider-config` and is in the `openshift-config` namespace.

File: modules/configuring-vsphere-connection-details.adoc
Comment (mareklibra) [inline modules/configuring-vsphere-connection-details.adoc:46]: You are right, its a pane, but `panel` would fit better here.

File: modules/configuring-vsphere-monitoring-configuration-completion.adoc
Comment (mareklibra) [inline modules/configuring-vsphere-monitoring-configuration-completion.adoc:9]: Not `at the end` but `during`

File: modules/configuring-vsphere-monitoring-configuration-completion.adoc
Comment (mareklibra) [inline modules/configuring-vsphere-monitoring-configuration-completion.adoc:27]:
```suggestion
. Check that the `PersistentVolumeClaims` can be bound
```

File: modules/configuring-vsphere-monitoring-configuration-completion.adoc
Comment (mareklibra) [inline modules/configuring-vsphere-monitoring-configuration-completion.adoc:29]: We might provide hints what values a `StorageClass` needs to contain to use the just configured connection.

See `Diagnostic Steps` in the KB: https://access.redhat.com/solutions/6677901

**Review by mareklibra — CHANGES_REQUESTED** [review]

File: modules/configuring-vsphere-connection-details.adoc
Comment (sgaoshang) [inline modules/configuring-vsphere-connection-details.adoc:36]: `that contains the virtual machines used to back-up the cluster` back-up is a little confused for me

File: modules/configuring-vsphere-connection-details.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-connection-details.adoc:28]: Do you know where the username is stored? Also in the kube-system namespace?

File: modules/configuring-vsphere-connection-details.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-connection-details.adoc:37]: Yes, I see now. At the time I think it was a dropdown, so that must have been a bug. Same with the next one.

File: modules/configuring-vsphere-connection-details.adoc
Comment (mareklibra) [reply modules/configuring-vsphere-connection-details.adoc:28]: The plugin stores them into secret name: `vsphere-creds`, namespace: `kube-system`.

File: modules/configuring-vsphere-monitoring-configuration-completion.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-monitoring-configuration-completion.adoc:30]: Let's decide on this together. Either I can just say 'using the following parameters' or I can write the GUI procedures out like in the Google Docs:
https://docs.google.com/document/d/1ES2Of_f584bO_D6o3msBYr-9kybEh3bVrnI4X-ZZ8s4/edit#

File: modules/configuring-vsphere-connection-details.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-connection-details.adoc]: @mareklibra Please could you explain this.

File: modules/configuring-vsphere-connection-details.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-connection-details.adoc]: I am assuming the password is also stored here, correct?

**Review by mareklibra — APPROVED** [review]

File: modules/configuring-vsphere-connection-settings.adoc
Comment (sgaoshang) [inline modules/configuring-vsphere-connection-settings.adoc:12]: suggest we put `vCenter address` here, I did not see `vCenter network address` in page vSphere connection configuration

File: modules/configuring-vsphere-connection-settings.adoc
Comment (sgaoshang) [inline modules/configuring-vsphere-connection-settings.adoc:15]: see above

File: modules/configuring-vsphere-connection-settings.adoc
Comment (sgaoshang) [inline modules/configuring-vsphere-connection-settings.adoc:18]: Suggest to change to `Virtual machine folder` in line with web page

File: modules/configuring-vsphere-connection-settings.adoc
Comment (sgaoshang) [inline modules/configuring-vsphere-connection-settings.adoc:22]: Doc shows `The cluster is connected to link:https://console.redhat.com/.` maybe we should remove the word `link:`

File: modules/configuring-vsphere-connection-settings.adoc
Comment (sgaoshang) [inline modules/configuring-vsphere-connection-settings.adoc:44]: `/SDDC-datacenter/vm/ci-ln-hjg4vg2-c61657-t2gzr` better to change SDDC-datacenter to SDDC-Datacenter in consistent with above example

File: modules/configuring-vsphere-connection-settings.adoc
Comment (sgaoshang) [inline modules/configuring-vsphere-connection-settings.adoc:33]: `vsphere-creds` is the secret name, suggest to change 
`The system will store the username and password into the `vsphere-creds` secret name`
to
`The system will store the username and password into the `vsphere-creds` secret`

File: modules/configuring-vsphere-connection-settings.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-connection-settings.adoc:44]: @mareklibra Please could you address this?

Comment (mareklibra) [reply modules/configuring-vsphere-connection-settings.adoc]: Back then, at the time of implementation, we were not considering multi-zones. I remember discussions happening afterwards.
@vrutkovs , do you know?

Comment (vrutkovs) [reply modules/configuring-vsphere-connection-settings.adoc]: Not aware of any multizone discussions. Its relatively new, so I suggest this document to mention single zone - and mention necessary changes in additional document

**Comment by tshwartz** [comment]
Hi @sgaoshang I really appreciate all your feedback and all issues have been addressed. Please approve if you have no further comments. Thanks!

File: modules/configuring-vsphere-checking-PVC-connections.adoc
Comment (kcarmichael08) [inline modules/configuring-vsphere-checking-PVC-connections.adoc:11]:
```suggestion
A persistent volume claim (PVC) is a request for storage by a user. Previously bound `PersistentVolumeClaims` objects can become disconnected as a result of the configuration. 
```

File: modules/configuring-vsphere-checking-PVC-connections.adoc
Comment (kcarmichael08) [inline modules/configuring-vsphere-checking-PVC-connections.adoc:13]:
```suggestion
To check that the connection to `PersistentVolumeClaims` objects still exists after configuration, create a `StorageClass` object, and then create a `PersistentVolumeClaim` object using that `StorageClass`. 
```

File: modules/configuring-vsphere-checking-PVC-connections.adoc
Comment (kcarmichael08) [inline modules/configuring-vsphere-checking-PVC-connections.adoc:3]: I actually don't see that this module is called in installing/installing_vsphere/installing-vsphere-post-installation-configuration.adoc unless I'm missing something. I don't see it in the HTML preview either?

File: modules/configuring-vsphere-verifying-configuration.adoc
Comment (kcarmichael08) [inline modules/configuring-vsphere-verifying-configuration.adoc:11]:
```suggestion
.Prerequisites
```

**Review by kcarmichael08 — COMMENTED** [review]
Hi, I marked some style issues and had a few other questions. Good job structuring the info!

**Comment by kelbrown20** [comment]
Hi Talia! It looks like there are some necessary nits, just some formating issues, objects that need backtics, and spelling errors, that Kerry brought up in her peer review that should be updated in this PR before this can be merged. 
Other than those things mentioned, this looks great!

**Comment by tshwartz** [comment]
Hi @kelbrown20 I didn't notice she'd left comments. Sorry about that!

File: modules/configuring-vsphere-verifying-configuration.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-verifying-configuration.adoc:25]: Changed to: Check that you are still able to bind `PersistentVolumeClaims` objects by performing the following steps:

File: modules/configuring-vsphere-connection-settings.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-connection-settings.adoc]: It is mandatory for Assisted Installer installations.

File: modules/configuring-vsphere-checking-PVC-connections.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-checking-PVC-connections.adoc:3]: I created two sections that I didn't use in the end and didn't delete this one.

File: modules/configuring-vsphere-checking-PVC-connections.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-checking-PVC-connections.adoc]: I deleted the whole section.

File: modules/configuring-vsphere-verifying-configuration.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-verifying-configuration.adoc]: @kcarmichael08 I changed it to _If you are encountering issues in creating the `PersistentVolumeClaims` object, you can troubleshoot by navigating to *Storage* -> *PersistentVolumeClaims* in the Web console._ Is that OK?

File: modules/configuring-vsphere-connection-settings.adoc
Comment (tshwartz) [reply modules/configuring-vsphere-connection-settings.adoc]: @kcarmichael08 Can I write _to store_ ? Future just makes more sense here. I've changed all the others.

**Comment by kcarmichael08** [comment]
New suggestions/ changes LGTM!

**Comment by tshwartz** [comment]
Thank you @kcarmichael08 !!!

/label merge-review-needed

**Review by kelbrown20 — COMMENTED** [review]
No worries, thank you for the updates! This looks great!

**Comment by kelbrown20** [comment]
@tshwartz It looks like auto cherrypick to 4.12 failed 😢 . I can merge the manual cherrypick whenever you're ready.

**Comment by tshwartz** [comment]
@kelbrown20 The new topic (bolded) needs to be added here in the map:
- Name: Installing a cluster on vSphere in a restricted network with user-provisioned infrastructure
   File: installing-restricted-networks-vsphere
**-Name: Configuring the vSphere connection settings after an installation [my new topic]
File: installing-vsphere-post-installation-configuration**
- Name: Uninstalling a cluster on vSphere that uses installer-provisioned infrastructure
   File: uninstalling-cluster-vsphere-installer-provisioned

**Pull request merged**
